DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/Delegates/DelegateProgressControllerTests.cs

```diff
@@ -138,6 +141,80 @@ string expectedAction
                 .WithActionName(expectedAction);
         }
 
+        [Test]
+        [TestCaseSource(
+            typeof(DelegateProgressControllerTests),
+            nameof(EditEndpointRedirectTestData)
+        )]
+        public void EditDelegateCourseAdminField_post_calls_service_and_redirects_to_correct_action(

```

Comment:
I don't understand what you mean by the redirect upon save not working. Surely this unit test is supposed to be the check for that, and it should not be passing if it's not fixed?

Reply:
So I just left this how the redirect tests work for the other Edit progress field actions, the redirection doesn't quite make sense as it is and will need to be fixed for all those actions as I described. I've started working on the fix, and it's easier than I thought so I may push it later today, but it will be a lot to test that will be rendered unnecessary if HEE decides to remove the Delegate Progress page.

Reply:
Not sure if I have missed a discussion somewhere, but what's the reason behind potentially getting rid of the Delegate Progress page? I understand that, because of the various shared details/links in Course Delegates and View Delegate pages, we have to incorporate different versions of the same page depending on how we access it. 

But I thought the Delegate Progress page was the main fix for it - funnel everything through the Delegate Progress page, so all details/pages are controlled from there. That way we only have to detail with two different sets of routes, and the pages remain easy to manage. Removing the Delegate Progress page would, I think, make the routing slightly more complex. We'd also have to move a whole bunch of pages (like the Learning Log and Detailed Progress pages) elsewhere.

Reply:
Here's the new ticket for the changes that are going to be made to the Delegate Progress page, it's going to be merged with the Detailed Progress page and it's still undecided whether we'll keep the edit buttons on the page or just on the cards. https://softwiretech.atlassian.net/browse/HEEDLS-862

Reply:
Cool, thanks for that.